scripts/test-truncating-of-client-hello.py

```diff
@@ -136,66 +131,72 @@ def main():
                                 ("hello truncate", -13, 0),
                                 ]:
 
-        conversation = Connect(host, port)
-        node = conversation
+        conversation_name = None
+        if "pad" in name:
+            conversation_name = "{0}: {1} of \"{2}\" byte padding".format(name, pad_len, pad_byte)
+        else:
+            conversation_name = "{0}: {1} of bytes truncated".format(name, -pad_len)
+        conversation = Conversation(conversation_name)
+
+        conversation.root_node = Connect(host, port)
         ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
-        node = node.add_child(pad_handshake(ClientHelloGenerator(ciphers,
+        node = conversation.root_node.add_child(pad_handshake(ClientHelloGenerator(ciphers,
                                                    extensions={ExtensionType.renegotiation_info: None}),
                                             pad_len, pad_byte))
         # we expect Alert and Close or just Close
         node = node.add_child(ExpectAlert())
         node.next_sibling = ExpectClose()
         node.add_child(ExpectClose())
+        conversations.append(conversation)
 
-        if "pad" in name:
-            conversations[name + ": " + str(pad_len) + " of \"" + str(pad_byte) +
-                          "\" byte padding"] = conversation
+    # first and last test is sanity
+    conversations.insert(0, conversation_sanity)
+    conversations.append(conversation_sanity)
+
+    # TBD: move it to separate function
+    for conversation in conversations:
+        if run_only and conversation.name not in run_only or conversation.name in run_exlude:
```

Comment:
corrected. I would like to move it to separate fn and call it from run_all fn because it's called every time after conversations compilation and before test run so it should be separated.